src/components/HoverHandler/index.js

```diff
@@ -2,34 +2,65 @@ import React, { PureComponent } from 'react'
 import PropTypes from 'prop-types'
 
 
+const DELAY = 300
+
+
 export default class HoverHandler extends PureComponent {
   static propTypes = {
     children: PropTypes.func.isRequired,
   }
 
   state = {
     hovering: false,
+    intent: false,
+  }
+
+  componentWillUnmount () {
+    if (this.timer) this.timer = clearTimeout(this.timer)
```

Comment:
since the component is unmounting, do you need to assign the value of `clearTimeout` to `this.timer`?

```javascript
componentWillUnmount () {
  if (this.timer) clearTimeout(this.timer)
}
```

Reply:
probably not